File: CSharp/GetRealGoodUtils.Tests/ObjectsTests.cs
using System;
using System.Configuration;
using NUnit.Framework;
using NUnit.Framework.Internal;

namespace GetRealGood.Utils.Tests
{
    
    
    public class ObjectsTests
    {
        
        
        
        [OneTimeSetUp]
        public static void SetupAll()
        {
            
            
            
        }

        [OneTimeTearDown]
        public static void TearDownAll()
        {
        }
        
        [SetUp]
        public void Setup()
        {
        }

        [TearDown]
        public void TearDown()
        {
        }

        class IsNullTests {
        
            [SetUp]
            public void Setup()
            {
            }

            [TearDown]
            public void TearDown()
            {
            }
            
        [Test]
        public void IsNull_Common_Object_Boolean()
        {
            
            object o = new object();
            Assert.AreEqual(Objects.IsNull(o), false);
        }

        [Test]
        public void IsNull_Nullable_Null_Boolean()
        {

            object o = null;
            Assert.AreEqual(Objects.IsNull(o), true);
        }
        }

        
        class IsNotNullTests
        {

            [SetUp]
            public void Setup()
            {
            }

            [TearDown]
            public void TearDown()
            {
            }

            [Test]
            public void IsNotNull_Common_Object_Boolean()
            {
                object o = new object();
                Assert.AreEqual(Objects.IsNotNull(o), true);
            }

            [Test]
            public void IsNotNull_Nullable_Null_Boolean()
            {
                object o = null;
                Assert.AreEqual(Objects.IsNotNull(o), false);
            }

        }

    }
}

File: Java/test/tech/getrealgood/util/test/ObjectsTests.java
package tech.getrealgood.util.test;

import org.junit.jupiter.api.*;
import org.junit.platform.commons.logging.Logger;
import org.junit.platform.commons.logging.LoggerFactory;
import tech.getrealgood.util.Objects;

import static org.junit.jupiter.api.Assertions.*;

class ObjectsTests {

    private static final Logger log = LoggerFactory.getLogger(ObjectsTests.class);

    @Nested
    class IsNull {

        @Test
        void isNull_Common_Object_Boolean() {
            Object o = new Object();
            assertEquals(false, Objects.isNull(o));
        }

        @Test
        void isNull_Nullable_Null_Boolean() {
            Object o = null;
            assertNotEquals(false, Objects.isNull(o));
        }
    }

    @Nested
    class IsNotNull {

        @Test
        public void isNotNull_Common_Object_Boolean() {
            Object o = new Object();
            assertEquals(true, Objects.isNotNull(o));
        }

        @Test
        public void isNotNull_Nullable_Null_Boolean() {
            Object o = null;
            assertNotEquals(true, Objects.isNotNull(o));
        }
    }

}